src/parse.rs

```diff
@@ -24,7 +24,8 @@ pub trait FromHex: Sized {
 
     /// Produces an object from a hex string.
     fn from_hex(s: &str) -> Result<Self, Self::Error> {
-        Self::from_byte_iter(HexToBytesIter::new(s)?)
+        let iter = HexToBytesIter::new(s).map_err(Into::into)?;
```

Comment:
When digging why this is not ambiguous I found that the whole API is likely broken around error handling. Will do more digging tomorrow.

Reply:
I tend to agree. I don't have an exact reason but all my recent attempts other than this one to improve the error handling of the `FromHex` trait have failed. Looking forward to seeing what you come up with.